tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2732,50 +2732,106 @@ class DoNotLogErrorsEndUpRaising(BaseChecker):
            Go through exception block and branches and ensure error hasn't been logged if exception is raised.
         """
-        # Return a list of exception blocks
-        except_block = node.handlers
-        # Iterate through each exception block
-        for nod in except_block:
-            # Get the nodes in each block (e.g. nodes Expr and Raise)
-            exception_block_body = nod.body
-            self.check_for_raise(exception_block_body)
-
-    def check_for_raise(self, node):
+        try:
+            # Return a list of exception blocks
+            except_block = node.handlers
+            # Iterate through each exception block
+            for nod in except_block:
+                # Get the nodes in each block (e.g. nodes Expr and Raise)
+                exception_block_body = nod.body
+                exception_name = nod.name.name
+                self.check_for_raise(exception_block_body, exception_name)
+        except:
+            pass
+
+    def check_for_raise(self, node, exception_name):
         """ Helper function - checks for instance of 'Raise' node
             Also checks 'If' and nested 'If' branches
         """
         for i in node:
             if isinstance(i, astroid.Raise):
-                self.check_for_logging(node)
+                self.check_for_logging(node, exception_name)
             # Check for any nested 'If' branches
             if isinstance(i, astroid.If):
-                self.check_for_raise(i.body)
+                self.check_for_raise(i.body, exception_name)
 
                 # Check any 'elif' or 'else' branches
-                self.check_for_raise(i.orelse)
+                self.check_for_raise(i.orelse, exception_name)
 
-    def check_for_logging(self, node):
-        """ Helper function - checks 'Expr' nodes to see if logging has occurred at 'warning' or 'error'
-            levels. Called from 'check_for_raise' function
+    def check_for_logging(self, node, exception_name):
+        """ Helper function - Called from 'check_for_raise' function
+            Checks if the same exception that's being raised is also being logged
         """
-        matches = [".warning", ".error", ".exception"]
-        for j in node:
-            if isinstance(j, astroid.Expr):
-                expression = j.as_string().lower()
-                if any(x in expression for x in matches):
-                    self.add_message(
-                        msgid=f"do-not-log-raised-errors",
-                        node=j,
-                        confidence=None,
-                    )
+
+        try:
+                
+            # Now check all logger calls to see if they're logging the exact same exception variable
+            for j in node:
+                if isinstance(j, astroid.Expr):
+                    for l in j.value.func.expr.infer():
+                        try:
+                            if "logger" in l.as_string():
```

Comment:
Is this looking at the name of the logger? How do we know that the logger is named `logger`?